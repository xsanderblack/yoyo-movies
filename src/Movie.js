import React from 'react';
import MoviePoster from './MoviePoster';
import ToggleMovieDetails from './ToggleMovieDetails';
import ToggleFav from './ToggleFav';

// Component renders movie information from props,
// and passes necessary data to ToggleFav and ToggleMovieDetails components.
const Movie = (props) => {
  const { id, title, poster_path, release_date } = props;
  return (
    <div>
      <MoviePoster size='thumbnail' path={poster_path} alt={title} />
      {title} - {release_date} - <ToggleFav {...props} />
      <ToggleMovieDetails id={id} />
    </div>
  );
};

export default Movie;
