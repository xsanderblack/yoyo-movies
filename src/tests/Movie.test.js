import React from 'react';
import ReactDOM from 'react-dom';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Movie from '../Movie';

it('Movie renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Movie />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should render Movie correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Movie />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
