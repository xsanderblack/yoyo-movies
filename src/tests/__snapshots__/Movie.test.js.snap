// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Movie correctly 1`] = `
<div>
  <MoviePoster
    size="thumbnail"
  />
   - 
   - 
  <ToggleFav />
  <ToggleMovieDetails />
</div>
`;
